chain/stmgr/stmgr.go

```diff
@@ -570,3 +570,21 @@ func (sm *StateManager) GetRandomnessDigestFromTickets(ctx context.Context, rand
 
 	return r.GetChainRandomness(ctx, randEpoch)
 }
+
+func (sm *StateManager) ResolveActorID(ctx context.Context, emitter address.Address, ts *types.TipSet) (abi.ActorID, error) {
+	var addr address.Address
+	if emitter.Protocol() == address.ID {
+		addr = emitter // already an ID address
+	} else {
+		var err error
+		addr, err = sm.LookupID(ctx, emitter, ts)
```

Comment:
If we introduce a new function, we should make a lookup ID use this new function instead of the other way around.

I would also prefer to do a pre-factor where we rename `LookupID` to `LookupIDAddress`.

Reply:
> we should make a lookup ID use this new function instead of the other way around.

This would mean translating address to ID and back again in `LookupID`, because the underlaying `StateTree.LookupID` API returns address (as well as state snapshot APIs). Right?

Reply:
> OK to refactor them too?

Yes, although it would be great if this could be done in a separate patch.

> This would mean translating address to ID and back again in LookupID, because the underlaying StateTree.LookupID API returns address (as well as state snapshot APIs). Right?

Ah, sorry, I mean the new `LookupIDAddress` function. That is, we take the current `LookupID` function and split it into a `LookupIDAddress` function and a `LookupID` function and have the `LookupIDAddress` address function called `LookupID` function.

Reply:
@Stebalien The changes you suggested above is causing enough churn across the code and diverges enough from the purpose of this PR that I prefer doing it in a separate PR. Unless you think it's a blocker for this one?

Reply:
Adressed here: https://github.com/filecoin-project/lotus/pull/11919